AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/StateMachine/hierarchical-state-machine-swift/StateMachine.swift

```diff
@@ -193,14 +138,12 @@ class StateMachine<
 
         subscribers[wrappedToken] = listener
 
-        onSubscribe?()
-
-        concurrentQueue.async {
+        Task {
             listener(currentState)
```

Comment:
By invoking listener async, I'm just now realizing that we are not guaranteeing the order of delivery: it would be possible for the Task to be deferred until after a new state machine transition has been processed. We should invoke `listener` synchronously from this method, right? (This was a bug in my original implementation, too, so hurray for desk checking!)

We should make sure we have tests that assert the order of listener invocation. I know we've got one below that asserts that we get the initial state, but we need to gain some confidence that we will always invoke the listener prior to processing the next actor message.

I'll need to update the sequence diagrams as well; they currently don't show that the SM invokes with current state when the subscription message is processed.